harness/determined/cli/template.py

```diff
@@ -42,14 +43,47 @@ def describe_template(args: Namespace) -> None:
 @authentication.required
 def set_template(args: Namespace) -> None:
     with args.template_file:
+        """
+        WARN: this downgrades the atomic behavior of upsert but it's
+        an acceptable tradeoff for now until we can remove this command.
+        """
+        session = cli.setup_session(args)
         body = util.safe_load_yaml_with_exceptions(args.template_file)
-        v1_template = bindings.v1Template(name=args.template_name, config=body)
-        bindings.put_PutTemplate(
-            cli.setup_session(args), template_name=args.template_name, body=v1_template
-        )
+        try:
+            bindings.get_GetTemplate(session, templateName=args.template_name).template
+            bindings.patch_PatchTemplateConfig(session, templateName=args.template_name, body=body)
+        except api.errors.NotFoundException:
+            v1_template = bindings.v1Template(name=args.template_name, config=body, workspaceId=0)
+            bindings.post_PostTemplate(session, template_name=args.template_name, body=v1_template)
         print(colored("Set template {}".format(args.template_name), "green"))
 
 
+@authentication.required
+def create_template(args: Namespace) -> None:
+    if not args.template_file:
+        raise ArgumentError(None, "template_file is required for set command")
+    body = util.safe_load_yaml_with_exceptions(args.template_file)
+    workspace_id = get_workspace_id_from_args(args) or 0
+    v1_template = bindings.v1Template(
+        name=args.template_name, config=body, workspaceId=workspace_id
+    )
+    bindings.post_PostTemplate(
+        cli.setup_session(args), template_name=args.template_name, body=v1_template
+    )
+    print(colored("Created template {}".format(args.template_name), "green"))
+
+
+@authentication.required
+def patch_tempalte_config(args: Namespace) -> None:
```

Comment:
```suggestion
def patch_template(args: Namespace) -> None:
```

Reply:
we should keep this in sync with whatever we do https://github.com/determined-ai/determined/pull/6605#discussion_r1182774437 I'll update if we decide to rename the cli cmd to just `template update`